src/lib.rs

```diff
@@ -225,14 +228,31 @@ impl Sheets {
         })
     }
 
+    /// Creates a new workbook from a reader.
+    /// ```
+    pub fn new(reader: RS, extension: &str) -> Result<Sheets<RS>> where RS: Read + Seek {
```

Comment:
I prefer to use an `enum` for the spreadsheet formats rather than filename extensions. We can map filename extensions to format variants in a separate function (if the `open` constructor is to remain).

Reply:
I agree it would probably be better

Reply:
I have a solution tested in practice, that works will. We can accept a Internet Media Type (‘MIME type’) instead of an extension.